Doc/whatsnew/3.8.rst

```diff
@@ -305,6 +305,12 @@ Changes in Python behavior
   in the leftmost :keyword:`for` clause).
   (Contributed by Serhiy Storchaka in :issue:`10544`.)
 
+* The compiler will emit a syntax warning when the ``is`` or ``is not``
+  operator is used with a literal (except singletons ``None``, ``False``,
```

Comment:
It is normally incorrect for code to perform an identity check against `True` or `False`.

Reply:
It is normal in the code that serializes Python data to external format that supports booleans. For example JSON, plist or pickle.

Since there are valid use cases for this, the compiler shouldn't emit a warning. Third-party checkers can do this if this check is optional.

Reply:
Identity checks against `True` and `False` are definitely code smells.  I always call them out in reviews.  But they are also fairly rare and as @serhiy-storchaka points out, do have some valid use cases.